File: PyzeWatchOSInAppObjC/PyzeWatchOSInAppObjC WatchKit Extension/NotificationController.m

#import "NotificationController.h"
#import <UserNotifications/UserNotifications.h>
#import <Pyze/Pyze.h>

@interface NotificationController()

@end


@implementation NotificationController

- (instancetype)init {
    self = [super init];
    if (self){
        // Initialize variables here.
        // Configure interface objects here.
        
    }
    return self;
}

- (void)willActivate {
    // This method is called when watch view controller is about to be visible to user
    [super willActivate];
}

- (void)didDeactivate {
    // This method is called when watch view controller is no longer visible
    [super didDeactivate];
}

#pragma mark - Handling Push notifications

#pragma mark - watchOS2

-(void)didReceiveRemoteNotification:(NSDictionary *)remoteNotification withCompletion:(void (^)(WKUserNotificationInterfaceType))completionHandler {
    
    NSDictionary *userInfo = [[remoteNotification objectForKey:@"aps"] objectForKey:@"alert"];
    NSString *title = [userInfo objectForKey:@"title"];
    NSString *message = [userInfo objectForKey:@"body"];
    
    WKAlertAction *act = [WKAlertAction actionWithTitle:@"OK"
                                                  style:WKAlertActionStyleCancel
                                                handler:^{
                                                    NSLog(@"Alert action 'OK' performed");
                                                }];
    NSArray *actions = @[act];
    
    [self presentAlertControllerWithTitle:title
                                  message:message
                           preferredStyle:WKAlertControllerStyleAlert
                                  actions:actions];
    
    [Pyze processReceivedRemoteNotification:userInfo];
    
}

#pragma mark - watchOS3

- (void)didReceiveNotification:(UNNotification *)notification withCompletion:(void(^)(WKUserNotificationInterfaceType interface)) completionHandler {
    
    NSDictionary *userInfo = [[notification.request.mutableCopy content] userInfo];
    NSDictionary *alertDict = [[userInfo objectForKey:@"aps"] objectForKey:@"alert"];
    NSString *title = [alertDict objectForKey:@"title"];
    NSString *message = [alertDict objectForKey:@"body"];
    
    
    WKAlertAction *act = [WKAlertAction actionWithTitle:@"OK"
                                                  style:WKAlertActionStyleCancel
                                                handler:^{
                                                    NSLog(@"Alert action 'OK' performed");
                                                }];
    NSArray *actions = @[act];
    
    [self presentAlertControllerWithTitle:title
                                  message:message
                           preferredStyle:WKAlertControllerStyleAlert
                                  actions:actions];
    
    [Pyze processReceivedRemoteNotification:userInfo];
    
    completionHandler(WKUserNotificationInterfaceTypeCustom);
}


@end

File: PyzeWatchOSInApp/PyzeWatchOSInApp WatchKit Extension/NotificationController.swift
import WatchKit
import Foundation
import UserNotifications
import Pyze


class NotificationController: WKUserNotificationInterfaceController {

    override init() {
        // Initialize variables here.
        super.init()
        
        // Configure interface objects here.
    }

    override func willActivate() {
        // This method is called when watch view controller is about to be visible to user
        super.willActivate()
    }

    override func didDeactivate() {
        // This method is called when watch view controller is no longer visible
        super.didDeactivate()
    }
    
    // MARK: - WatchOS2
    
    override func didReceiveRemoteNotification(_ remoteNotification: [AnyHashable : Any], withCompletion completionHandler: @escaping (WKUserNotificationInterfaceType) -> Void) {
        
        if let aps = remoteNotification["aps"] as? NSDictionary {
            if let alert = aps["alert"] as? NSDictionary {
                if let title = alert["title"] as? NSString {
                    if let body = alert["body"] as? NSString {
                        
                        
                        let alert:WKAlertAction = WKAlertAction(title: "OK", style: .default, handler: {
                            print("Alert action 'OK' performed")
                        })
                        let actions = [alert]
                        self.presentAlert(withTitle: title as String, message: body as String, preferredStyle: .alert, actions: actions)
                        
                        Pyze.processReceivedRemoteNotification(remoteNotification)
                    }
                }
            }
        }
        
    }
    

    // MARK: - WatchOS3
    
    @available(watchOSApplicationExtension 3.0, *)
    override func didReceive(_ notification: UNNotification, withCompletion completionHandler: @escaping (WKUserNotificationInterfaceType) -> Swift.Void) {
        
        
        let userInfo = notification.request.content.userInfo
        
        if let aps = userInfo["aps"] as? NSDictionary {
            if let alert = aps["alert"] as? NSDictionary {
                if let title = alert["title"] as? NSString {
                    if let body = alert["body"] as? NSString {
                        
                        let alert:WKAlertAction = WKAlertAction(title: "OK", style: .default, handler: {
                            print("Alert action 'OK' performed")
                        })
                        let actions = [alert]
                        self.presentAlert(withTitle: title as String, message: body as String, preferredStyle: .alert, actions: actions)
                        
                        
                        Pyze.processReceivedRemoteNotification(userInfo)
                    }
                }
            }
        }
        
        
        completionHandler(.custom)
    }
    
}
